lib/include/pl/core/ast/ast_node_bitfield_array_variable_decl.hpp

```diff
@@ -0,0 +1,206 @@
+                if (startArrayIndex.has_value())
+                    evaluator->setCurrentArrayIndex(*startArrayIndex);
+                else
+                    evaluator->clearCurrentArrayIndex();
+            };
+
+            auto arrayPattern = std::make_unique<ptrn::PatternBitfieldArray>(evaluator, evaluator->dataOffset(), 0, evaluator->getBitfieldBitOffset(), 0);
+            arrayPattern->setVariableName(this->m_name);
+            arrayPattern->setSection(evaluator->getSectionId());
+
+            std::vector<std::shared_ptr<ptrn::Pattern>> entries;
+
+            size_t size = 0;
+            u128 entryIndex = 0;
+
+            auto addEntries = [&](std::vector<std::shared_ptr<ptrn::Pattern>> &&patterns) {
+                for (auto &pattern : patterns) {
+                    pattern->setVariableName(fmt::format("[{}]", entryIndex));
+                    pattern->setEndian(arrayPattern->getEndian());
+                    if (pattern->getSection() == ptrn::Pattern::MainSectionId)
+                        pattern->setSection(arrayPattern->getSection());
+
+                    size += pattern->getSize();
+                    entryIndex++;
+
+                    entries.push_back(std::move(pattern));
+
+                    evaluator->handleAbort();
+                }
+            };
+
+            if (this->m_size != nullptr) {
+                auto sizeNode = this->m_size->evaluate(evaluator);
+                std::variant<u128, ASTNodeWhileStatement *> boundsCondition;
+
+                if (auto literal = dynamic_cast<ASTNodeLiteral *>(sizeNode.get()); literal != nullptr) {
+                    boundsCondition = std::visit(wolv::util::overloaded {
+                        [this](const std::string &) -> u128 { err::E0006.throwError("Cannot use string to index array.", "Try using an integral type instead.", this); },
+                        [this](ptrn::Pattern *pattern) -> u128 {err::E0006.throwError(fmt::format("Cannot use custom type '{}' to index array.", pattern->getTypeName()), "Try using an integral type instead.", this); },
+                        [](auto &&size) -> u128 { return size; }
+                    }, literal->getValue());
+                } else if (auto whileStatement = dynamic_cast<ASTNodeWhileStatement *>(sizeNode.get()); whileStatement != nullptr) {
+                    boundsCondition = whileStatement;
+                } else {
+                    err::E0001.throwError(fmt::format("Unexpected type of bitfield array size node."), {}, this);
+                }
+
+                auto limit = evaluator->getArrayLimit();
+                auto checkLimit = [&](auto count) {
+                    if (count > limit)
+                        err::E0007.throwError(fmt::format("Bitfield array grew past set limit of {}", limit), "If this is intended, try increasing the limit using '#pragma array_limit <new_limit>'.", this);
+                };
+
+                if (std::holds_alternative<u128>(boundsCondition))
```

Comment:
You probably want to use `boundsCondition.toUnsigned()` here instead because it's valid to use signed numbers, chars or bools (or anything that can convert to a number really)

Reply:
The conversion happens above here at lines 118-122, I believe it should accept any of those types as that `std::visit()` call accepts the same types as `toUnsigned()`.